### PR TITLE
Added Flow in src/actions/ast/setInScopeLines.js

### PR DESCRIPTION
**Pull request by Kanikashrivastav**
Fixes a part #5120

### CODE REVIEW

File: src/actions/ast/setInScopeLines.js [hunk 1]
```diff
@@ -2,18 +2,22 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at <http://mozilla.org/MPL/2.0/>. */
 
+// @flow
+
 import { getOutOfScopeLocations, getSelectedSource } from "../../selectors";
 import { getSourceLineCount } from "../../utils/source";
 
 import { range, flatMap, uniq, without } from "lodash";
+import type { AstLocation } from "../../workers/parser";
```
Comment (darkwing) [inline src/actions/ast/setInScopeLines.js]: These probably aren't required.

File: src/actions/ast/setInScopeLines.js [hunk 2]
```diff
@@ -2,18 +2,22 @@
...
+import type { ThunkArgs } from "../types";
 
 function getOutOfScopeLines(outOfScopeLocations: AstLocation[]) {
   if (!outOfScopeLocations) {
     return null;
   }
 
   return uniq(
-    flatMap(outOfScopeLocations, location =>
+    flatMap((outOfScopeLocations:any), location =>
```
Comment (darkwing) [inline src/actions/ast/setInScopeLines.js]: We should avoid `:any` and use its actual type.

File: src/actions/ast/setInScopeLines.js [hunk 3]
```diff
@@ -2,12 +2,15 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at <http://mozilla.org/MPL/2.0/>. */
 
+// @flow
+
 import { getOutOfScopeLocations, getSelectedSource } from "../../selectors";
 import { getSourceLineCount } from "../../utils/source";
 
 import { range, flatMap, uniq, without } from "lodash";
+import type { ThunkArgs } from "../types";
```
Comment (darkwing) [inline src/actions/ast/setInScopeLines.js]: I don't think this is necessary.

File: src/actions/ast/setInScopeLines.js [hunk 4]
```diff
@@ -2,12 +2,15 @@
...
 
-function getOutOfScopeLines(outOfScopeLocations: AstLocation[]) {
+function getOutOfScopeLines(outOfScopeLocations) {
```
Comment (darkwing) [inline src/actions/ast/setInScopeLines.js]: Why remove the `AstLocation` type here?

**Comment by Kanikashrivastav** [comment]
I am getting these error if I do the changes that you suggested, I have installed everything using 

> yarn install

 
image
image

**Comment by darkwing** [comment]
Right -- you'll need to import that type.

**Comment by Kanikashrivastav** [comment]
So, shouldn't this be merged rather than closed? @darkwing

**Comment by darkwing** [comment]
I was hoping you could take another shot at this @Kanikashrivastav; the changes here are going in the wrong direction and could use a fresh start.

**Pull request closed without merging**